dds/idl/typeobject_generator.cpp

```diff
@@ -336,12 +396,31 @@ typeobject_generator::gen_union(AST_Union*, UTL_ScopedName* name,
     const string decl_gto = "getMinimalTypeObject<" + clazz + ">";
     Function gto(decl_gto.c_str(), "const XTypes::TypeObject&", "");
     gto.endArgs();
-
+    const ExtensibilityKind exten = be_global->extensibility(node);
+    string type_flag_str;
+    switch (exten) {
+      case extensibilitykind_final:
+        type_flag_str = "XTypes::IS_FINAL";
+        break;
+      case extensibilitykind_appendable:
+        type_flag_str = "XTypes::IS_APPENDABLE";
+        break;
+      case extensibilitykind_mutable:
+        type_flag_str = "XTypes::IS_MUTABLE";
+        break;
+      default:
+        idl_global->err()->misc_error(
+          "Unexpected extensibility while setting flags", node);
+        return false;
+    }
```

Comment:
Since you needed more of these switches, I would consider turning this into a function.